Assets/Mirror/Runtime/NetworkAuthenticator.cs

```diff
@@ -0,0 +1,71 @@
+using System;
+using UnityEngine;
+using UnityEngine.Events;
+
+namespace Mirror
+{
+    /// <summary>
+    /// Unity Event for the NetworkConnection
+    /// </summary>
+    [Serializable] public class UnityEventNetworkConnection : UnityEvent<NetworkConnection> { }
+
+    /// <summary>
+    /// Base class for implementing component-based authentication during the Connect phase
+    /// </summary>
+    [AddComponentMenu("Network/NetworkAuthenticator")]
+    [HelpURL("https://mirror-networking.com/xmldocs/articles/Concepts/Authentication.html")]
+    public class NetworkAuthenticator : MonoBehaviour
```

Comment:
one more thing.
with this implementation, we add an empty NetworkAuthenticator to every single project.
it's not that obvious that people have to remove that template and replace it with an authenticator that inherits from it.

why don't we:
* make NetworkAuthenticator abstract
* in Networkmanager check if we have one, otherwise set conn.isAuthenticated by default

this way everything still works as before, and people that want authenticators can add them as needed.

imho it's also easier to use. people simply inherit from an abstract NetworkAuthenticator and add it to their NetworkManager, done.

Reply:
The NetworkAuthenticator component isn't empty, and When I add a configurable timeout that will appear as a slider above the event lists.

![image](https://user-images.githubusercontent.com/9826063/64922682-e8fa4b00-d79f-11e9-9615-a2b68c3c8b43.png)

Reply:
It is the same as transports,  so I don't see why it would be confusing.

Reply:
The little Help button on the component takes them directly to a doc that tells them exactly how to make their own, how to swap it, and includes a fully working example.  Under the hood, the default NetworkAuthenticator methods aren't empty either.  ServerAuthenticate and ClientAuthenticate both have code in them so devs can call the base methods from their overrides or anywhere else in their inherited class when their authentication process ends in success.

Reply:
I will try a version with abstract auth first